From e28a3464c66b5d137de4acb6595ff56a1da7d149 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 25 Oct 2017 09:20:56 +0100
Subject: [PATCH 124/454] scripts: Update mkknlimg, just in case

With the removal of the vc_cma driver, mkknlimg lost an indication that
the user had built a downstream kernel. Update the script, adding a few
more key strings, in case it is still being used.

Note that mkknlimg is now deprecated, except to tag kernels as upstream
(283x), and thus requiring upstream DTBs.

See: https://github.com/raspberrypi/linux/issues/2239

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 scripts/mkknlimg | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

--- a/scripts/mkknlimg
+++ b/scripts/mkknlimg
@@ -68,18 +68,15 @@ if (! -r $kernel_file)
 
 my $wanted_strings =
 {
-	'bcm2708_fb' => FLAG_PI | FLAG_270X,
 	'brcm,bcm2835-mmc' => FLAG_PI,
 	'brcm,bcm2835-sdhost' => FLAG_PI,
-	'brcm,bcm2708-pinctrl' => FLAG_PI | FLAG_DTOK,
 	'brcm,bcm2835-gpio' => FLAG_PI | FLAG_DTOK,
-	'brcm,bcm2708' => FLAG_PI | FLAG_DTOK | FLAG_270X,
-	'brcm,bcm2709' => FLAG_PI | FLAG_DTOK | FLAG_270X,
+	'brcm,bcm2708-fb'   => FLAG_PI | FLAG_DTOK | FLAG_270X,
+	'brcm,bcm2708-usb'  => FLAG_PI | FLAG_DTOK | FLAG_270X,
 	'brcm,bcm2835' => FLAG_PI | FLAG_DTOK | FLAG_283X,
 	'brcm,bcm2836' => FLAG_PI | FLAG_DTOK | FLAG_283X,
+	'brcm,bcm2837' => FLAG_PI | FLAG_DTOK | FLAG_283X,
 	'of_cfs_init' => FLAG_DTOK | FLAG_DDTK,
-	'vc_cma_init' => FLAG_PI | FLAG_270X,
-	'vc-mem' => FLAG_PI | FLAG_270X,
 };
 
 my $res = try_extract($kernel_file, $tmpfile1);
